apps/app_dial.c

```diff
@@ -1686,6 +1687,19 @@ static struct ast_channel *wait_for_answer(struct ast_channel *in,
 				case AST_CONTROL_PVT_CAUSE_CODE:
 					ast_indicate_data(in, AST_CONTROL_PVT_CAUSE_CODE, f->data.ptr, f->datalen);
 					break;
+				case AST_CONTROL_PLAYBACK_BEGIN:
+					filename = pbx_builtin_getvar_helper(c, "DIALING_PLAYBACK_ANNOUNCEMENT");
```

Comment:
The filename should be a data payload on the frame, and not just put as a dialplan variable.